Use ST_Resample for prescaling

strk opened the pull request:
Ensures all tiles are properly aligned.
Fixes #2375 (visible tile boundaries).

strk commented:
BTW, the change also speeds up selection:
```
gjnzhkasuc=# explain analyze SELECT ST_AsBinary(ST_Resample("rast", 3792.98::float8, -3792.98::float8)) AS geom FROM file_wnbnfkkqshjdrhpkdchn;    
                                                       QUERY PLAN                                                       
------------------------------------------------------------------------------------------------------------------------
 Seq Scan on file_wnbnfkkqshjdrhpkdchn  (cost=0.00..4.05 rows=70 width=258) (actual time=0.958..28.890 rows=70 loops=1)
 Planning time: 0.486 ms
 Execution time: 28.957 ms
(3 rows)

gjnzhkasuc=# explain analyze SELECT ST_AsBinary(ST_Resize("rast", least(abs(ST_ScaleX("rast"))::float8/3792.98, 1.0), least(abs(ST_ScaleY("rast"))::float8/3792.98, 1.0))) AS geom FROM file_wnbnfkkqshjdrhpkdchn;
                                                       QUERY PLAN                                                        
-------------------------------------------------------------------------------------------------------------------------
 Seq Scan on file_wnbnfkkqshjdrhpkdchn  (cost=0.00..22.43 rows=70 width=258) (actual time=1.455..43.418 rows=70 loops=1)
 Planning time: 0.189 ms
 Execution time: 43.501 ms
(3 rows)
```

strk commented:
Build is failing as it is failing in master too. Should I rebase this PR to a more stable branch ?

artemp commented:
@strk - At least one is green : https://travis-ci.org/mapnik/mapnik/jobs/113683240 :)

Ready to push once warning is fixed : https://travis-ci.org/mapnik/mapnik/jobs/113683240#L1754 
Thanks!

strk commented:
Warning fixed

strk commented:
Where had pgraster_test.py gone, btw ? I think it would be worth guarding such change with a testcase

artemp commented:
@strk - thanks! The test is still there https://github.com/mapnik/python-mapnik/blob/master/test/python_tests/pgraster_test.py but we don't package bindings into core anymore.
Ideally we need a c++ unit test in https://github.com/mapnik/mapnik/tree/master/test/unit/datasource !

strk commented:
Is it ok to piggy-back postgis.cpp, being at least the initialization
all the same ?

strk commented:
Forget it, I'll just copy it and use as a starting point
for a pgraster.cpp

artemp commented:
@strk - of course :+1:

strk commented:
So I just pushed a lame test copied from postgis.cpp into pgraster.cpp.
The test is very lame, doesn't really check anything interesting as the python version does.
And also it doesn't test for the visible-tile bug at all.

strk commented:
Counter-order. The tile boundaries are still visible with the patch, so the change was not enough.

strk commented:
I'm closing this as not solving the issue. Refer to #3358 for the sole testcase.

strk commented:
Reopening as after all there's still some value in switching to ST_Resample, even if it doesn't fix the visible tile thing.

strk commented:
Squash-rebased against master, and removed test (now available as #3358).
The scope of this PR was also changed from a bugfix to an enhancement (see commit log).

strk commented:
Let me know if you prefer that I close this PR and create a new one, for clarity (as the initial PR description is now misleading)
